ospfd/ospf_zebra.c

```diff
@@ -1365,6 +1613,15 @@ void ospf_zebra_init(struct thread_master *master, unsigned short instance)
 	zclient->redistribute_route_add = ospf_zebra_read_route;
 	zclient->redistribute_route_del = ospf_zebra_read_route;
 
+	/* Initialize special zclient for synchronous message exchanges. */
+	struct zclient_options options = zclient_options_default;
+	options.synchronous = true;
+	zclient_sync = zclient_new(master, &options);
+	zclient_sync->sock = -1;
+	zclient_sync->redist_default = ZEBRA_ROUTE_OSPF;
+	zclient_sync->instance = instance;
```

Comment:
Now that #5925 was merged, a better solution to this "instance collision problem" would be to add a new `synchronous` parameter to `zserv_find_client()` and adapt all call sites accordingly (only the LM code should look for synchronous zclients). That would be more robust in my opinion.

Reply:
I'm not sure that works - the bgp LM client code is not synchronous, is it? I think it would be clearer to make the support for multiple client sessions per daemon explicit, by adding an optional session id to zclient. the default version of `zserv_find_client` wouldn't have to change, but we could add a version that took an extra value. the only place that'd need to be plumbed through would be the LM code.

Reply:
@mjstapp Make a proposal with PR #6222 Please comment this new PR.

Reply:
@mjstapp indeed, good observation. Your proposal to add a new "session id" to zclient sounds good to me.